migration-engine/connectors/migration-connector/src/steps.rs

```diff
@@ -37,368 +35,266 @@ where
-        self.tpe == index_definition.tpe && self.fields == index_definition.fields
+impl Into<ast::Argument> for &Argument {
+    fn into(self) -> ast::Argument {
+        ast::Argument {
+            name: ast::Identifier::new(&self.name),
+            value: self.value.to_ast_expression(),
+            span: ast::Span::empty(),
+        }
     }
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[test]
-    fn delete_index_must_apply_to_the_right_indexes() {
-        let definition = IndexDefinition {
-            fields: vec!["testColumn".into()],
-            tpe: IndexType::Unique,
-            name: None,
-        };
-        let correct_delete_index = DeleteIndex {
-            model: "ignored".into(),
-            fields: vec!["testColumn".into()],
-            tpe: IndexType::Unique,
-            name: None,
-        };
-
-        assert!(correct_delete_index.applies_to_index(&definition));
-
-        let delete_index = DeleteIndex {
-            tpe: IndexType::Normal,
-            ..correct_delete_index.clone()
-        };
-
-        // tpe does not match
-        assert!(!delete_index.applies_to_index(&definition));
-
-        let delete_index = DeleteIndex {
-            name: Some("index_on_testColumn".to_owned()),
-            ..correct_delete_index.clone()
-        };
-
-        // name does not match, but it does not matter
-        assert!(delete_index.applies_to_index(&definition));
+#[derive(Debug, Deserialize, Serialize, PartialEq, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct DirectiveLocator {
+    #[serde(flatten)]
+    pub location: DirectiveLocation,
+    pub r#directive: String,
+    /// The arguments of the directive are required to match directives that can be repeated,
+    /// like `@@unique` on a model. This is `None` when matching can be done without comparing
+    /// the arguments, and `Some` when a directive should be matched exactly.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub arguments: Option<Vec<Argument>>,
+}
 
-        let delete_index = DeleteIndex {
-            fields: vec!["testColumn".into(), "otherColumn".into()],
-            ..correct_delete_index.clone()
-        };
+impl DirectiveLocator {
+    pub fn matches_ast_directive(&self, directive: &ast::Directive) -> bool {
+        if self.directive != directive.name.name {
+            return false;
+        }
 
-        // fields do not match
-        assert!(!delete_index.applies_to_index(&definition));
+        match &self.arguments {
+            Some(arguments) => {
+                directive.arguments.len() == arguments.len()
+                    && directive
+                        .arguments
+                        .iter()
+                        .zip(arguments.iter())
+                        .all(|(directive_argument, self_argument)| {
+                            self_argument.matches_ast_argument(directive_argument)
+                        })
+            }
```

Comment:
TODO: this is sensitive to reordering of the arguments, we should make it order-insensitive